include/inviwo/core/properties/property.h

```diff
@@ -281,6 +282,44 @@ class IVW_CORE_API Property : public PropertyObservable, public Serializable, pu
     virtual void setVisible(bool val);
     virtual bool getVisible();
 
+    /* \brief sets visibility depending another property `prop`, according to `callback`
+     * @param prop is the property on which the visibility depends
+     * @param callback is a function that outputs a visibility boolean value. The function gets `prop` as parameter
+     *
+     * Checks the expression in `callback` every time `prop` is changed and sets the
+     * visibility accordingly. Note that this registers an onChange callback on `prop`, which
+     * might result in poor performance when `prop` is a very frequently changed property.
+     */
+    template<typename P, typename DecisionFunc>
+    const BaseCallBack* visibilityDependsOn(P& prop, DecisionFunc callback) {
```

Comment:
Should this also immediately call lambda to set the visibility so the states are in sync?